bootstrap/kubeadm/api/v1alpha4/kubeadmconfig_webhook.go

```diff
@@ -115,6 +115,11 @@ func (c *KubeadmConfigSpec) validate(name string) error {
 		knownPaths[file.Path] = struct{}{}
 	}
 
+	if c.Ignition != nil && c.Format != Ignition {
```

Comment:
Should we add some more validation rule?
Eg, IgnitionSpec spec could not be != nil if we are not using the ignition format.
Also, should we validate that in case if format = ignition we are not setting unsupported combinations of other fields (ntp, users, etc)

Reply:
>Also, should we validate that in case if format = ignition we are not setting unsupported combinations of other fields (ntp, users, etc)

Ah, thanks for reminding me about this. I was planning to eventually implement support for them. But initially we can reject them.

Reply:
I added the rejection code for now.

Reply:
@fabriziopandini about this validation, users usually create `KubeadmControlplane` and `KubeadmConfigTemplate`, not `KubeadmConfig` directly. Do you think those objects should also have a webhook implementation, so when I apply the cluster manifests, I immediately get feedback, that my data is wrong? Otherwise you still have to check the logs of the controllers to figure out that your template is wrong...

Reply:
>I added the rejection code for now.

There is actually more code to be added there, as some of the `Files` features are not implemented either. I'm looking into that now.

Reply:
> users usually create KubeadmControlplane and KubeadmConfigTemplate

Now that we are doing some validations, this makes definitely sense for me. Let's make sure to reuse the same code for validation checks.

> There is actually more code to be added there, as some of the Files features are not implemented either. I'm looking into that now.

Thanks for that, appreciated.
Up to you to implement this in this PR or open an issue to track the problem + a separated PR (so we are not conflating unrelated issues in the scope of this PR)

Reply:
>Now that we are doing some validations, this makes definitely sense for me. Let's make sure to reuse the same code for validation checks.

I'll look into that :+1: Thanks!

>There is actually more code to be added there, as some of the Files features are not implemented either. I'm looking into that now.

This is now implemented :)

Reply:
Would it be fine to expose some sort of `Validate()` method on `KubeadmConfigSpec`, so the code can be re-used in `KubeadmControlPlane` webhook? Right now those 2 types live in different packages, so code cannot be shared without exposing it.

Reply:
+1 for me

Reply:
Please add unit test for validations